smart-contracts/assembly/contracts/registry.ts

```diff
@@ -63,58 +92,69 @@ export function createNewPool(binaryArgs: StaticArray<u8>): void {
     .nextString()
     .expect('TokenAddress B is missing or invalid');
 
-  const feeShareProtocol = args
-    .nextF64()
-    .expect('ProtocolFee is missing or invalid');
-
   const inputFeeRate = args
     .nextF64()
     .expect('InputFeeRate is missing or invalid');
 
+  // Ensure that the fee share protocol is between 0 and 1
+  assert(
+    isBetweenZeroAndOne(inputFeeRate),
+    'Fee share protocol must be between 0 and 1',
+  );
+
+  // Ensure taht the aTokenAddress and bTokenAddress are smart contract addresses
```

Comment:
Check for Massa token if also smart contract.
Is an NFT a smart contract? Would this test pass?